scenarios/mixed_arc_get_agent_activity/src/main.rs

```diff
@@ -0,0 +1,190 @@
+    record_open_connections_if_necessary(ctx)?;
+
+    let _: ActionHash = call_zome(
+        ctx,
+        "agent_activity",
+        "create_sample_entry",
+        "this is a test entry value",
+    )?;
+
+    ctx.get_mut().scenario_values.entries_created_count += 1;
+
+    let agent_pub_key = ctx.get().cell_id().agent_pubkey().to_string();
+    let metric = ReportMetric::new("mixed_arc_get_agent_activity_entry_created_count")
+        .with_tag("agent", agent_pub_key)
+        .with_tag("behaviour", ctx.assigned_behaviour().to_string())
+        .with_field("value", ctx.get().scenario_values.entries_created_count);
+    ctx.runner_context().reporter().clone().add_custom(metric);
+
+    Ok(())
+}
+
+fn agent_behaviour_get_agent_activity(
+    ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<ScenarioValues>>,
+) -> HookResult {
+    record_open_connections_if_necessary(ctx)?;
+
+    let reporter = ctx.runner_context().reporter();
+
+    match ctx.get().scenario_values.write_peer.clone() {
+        Some(write_peer) => {
+            let activity_result: anyhow::Result<AgentActivity> = call_zome(
+                ctx,
+                "agent_activity",
+                "get_agent_activity_full",
+                write_peer.clone(),
+            );
+
+            let agent_pub_key = ctx.get().cell_id().agent_pubkey().to_string();
+
+            if let Ok(activity) = activity_result {
+                let previous_observed_chain_head =
+                    ctx.get().scenario_values.last_observed_chain_head;
+
+                let highest_observed = activity
+                    .highest_observed
+                    .map_or(previous_observed_chain_head.0, |v| v.action_seq);
+
+                // If it's a new chain head, record the time difference between the new one and the previous one
+                // (or divide by n if there are more than one chain head since).
+                // Note that the chain head may decrease again in case the agent activity gets fetched
+                // from a different peer than in the previous invocation of the behaviour, therefore
+                // a sturating subtraction is needed here.
+                let n = highest_observed.saturating_sub(previous_observed_chain_head.0);
+                let now = Timestamp::now().0;
+                if n > 0 {
+                    let time_delta_s = (now as f64 - previous_observed_chain_head.1 as f64) / 1e6;
+                    reporter.add_custom(
+                        ReportMetric::new("mixed_arc_get_agent_activity_new_chain_head_delay")
+                            .with_field("value", time_delta_s),
+                    );
+                }
+
+                reporter.add_custom(
+                    ReportMetric::new("mixed_arc_get_agent_activity_highest_observed_action_seq")
+                        .with_tag("get_agent_activity_agent", agent_pub_key)
+                        .with_tag("write_agent", write_peer.to_string())
```

Comment:
Do we want to tag if the write_agent was a zero-arc or full-arc?

Reply:
I wanted to do that too first but we can't determine that since as the reading agent we only know the agent pubkey of the write agent...